src/Microsoft.VisualStudio.Editor.Razor/DefaultVisualStudioRazorParser.cs

```diff
@@ -96,6 +96,8 @@ internal DefaultVisualStudioRazorParser(RazorCodeDocument codeDocument)
 
         public override ITextBuffer TextBuffer => _documentTracker.TextBuffer;
 
+        public override bool IsIdle => _pendingStructureChanges == 0 && _parser.IsIdle;
```

Comment:
WTE code is highly coupled to the idea that they can query their RazorCodeGenerator for its "readiness" and react in different ways based on the outcome. Prior to this change we'd notify WTE on the background thread and they'd determined readiness by when a document structure change was processed on the foreground thread. Now that we dispatch on the foreground thread, so they have less ceremony, we have to provide them a "readiness" replacement.

I would have rathered an "ensure ready" async equivalent to avoid all the IsReady ceremony but from my investigation it looks like that'd require a significant amount of WTE changes. I'm open to other ideas here if you have any.

Reply:
Can this be called `HasPendingChanges` or something similar?

Seeing code like:
```
public override bool IsIdle => _pendingStructureChanges ...
```

Sadly off my *badly-designed-contept* early warning system

Reply:
I want to understand exactly what it they do that's stateful here. I think the code changes are fine, I just want to learn more

Reply:
> I want to understand exactly what it they do that's stateful here.

Not sure I understand the ask. One example of them being stateful based off of the result of this property is they determine ranges in the document that they call "artifacts". Those artifacts are stateful and get updated whenever document structure changes flow through the system.

> Can this be called HasPendingChanges or something similar?

Yup